pkg/app/app.go

```diff
@@ -210,6 +222,17 @@ func (app *App) setupRepo() (bool, error) {
 		}
 	}
 
+	// Run this afterward so that the previous repo creation steps can run without this interfering
+	if isBare, err := isBareRepo(app.OSCommand); isBare {
+		if err != nil {
+			return false, err
+		}
+
+		if isBare {
+			log.Fatalln("bare repositories are not supported by lazygit, please make this a working repository.")
```

Comment:
I'd add the word `yet` here somewhere to indicate that it _is_ on the roadmap, but hasn't been implemented as of this moment.